pkg/volume/host_path/host_path.go

```diff
@@ -215,6 +231,44 @@ func (r *hostPathRecycler) Recycle() error {
 	return volume.RecycleVolumeByWatchingPodUntilCompletion(pod, r.host.GetKubeClient())
 }
 
+// hostPathCreater implements a dynamic provisioning Creater for the HostPath plugin
+// This implementation is meant for testing only and only works in a single node cluster.
+type hostPathCreater struct {
+	host    volume.VolumeHost
+	options volume.VolumeOptions
+}
+
+// Create for hostPath simply creates a local /tmp/hostpath_pv/%s directory as a new PersistentVolume.
+// This Creater is meant for development and testing only and WILL NOT WORK in a multi-node cluster.
+func (r *hostPathCreater) Create() (*api.PersistentVolume, error) {
+	fullpath := fmt.Sprintf("/tmp/hostpath_pv/%s", util.NewUUID())
+	err := os.MkdirAll(fullpath, 0750)
```

Comment:
As HostPath provisioning is only applicable to single node, most likely dev clusters. How do you feel about changing this to 0777? Any pods running as non-root can't use HostPath provisioned volumes otherwise. Thinking specifically about minikube/minishift which would find a usable HostPath provisioner invaluable.